[FIX] sale_order_product_recommendation: AccessError when getting recommendations

yajo opened the pull request:
A simple salesperson couldn't access recommendations after https://github.com/OCA/sale-workflow/pull/2913 was merged.

It turns out that these tests were not using the testing user they created, although it seemed like so.

Now I'm reducing the permissions for the user that runs the tests, and thus can fix the issue and get it tested.

@moduon MT-4941

yajo committed:
[FIX] sale_order_product_recommendation: AccessError when getting recommendations

A simple salesperson couldn't access recommendations after https://github.com/OCA/sale-workflow/pull/2913 was merged.

It turns out that these tests were not using the testing user they created, although it seemed like so.

Now I'm reducing the permissions for the user that runs the tests, and thus can fix the issue and get it tested.

@moduon MT-4941

fcvalgar commented:
Hi @yajo ,

The recommender does not launch when I select the recommender options.

https://www.loom.com/share/b4648db83fe049e1af25917497dc03c7?sid=382b930a-7205-4877-8e8f-3f19ba22b8f4

Thank you.

yajo committed:
[FIX] sale_order_product_recommendation: feedback when nothing is found

As https://github.com/OCA/sale-workflow/pull/2936#issuecomment-1918785798 shows, when there's nothing found and there's no user feedback, the wizard looked like stuck.

Now there's a visual feedback indicating the user what's happening and what to do.

@moduon MT-4941

yajo commented:
Please try again.

fcvalgar reviewed (approved):
Functional review, 

LGTM, thank you @yajo 

![image](https://github.com/OCA/sale-workflow/assets/17847432/77b2ef9c-8971-45b4-8320-a4d694713cab)


![image](https://github.com/OCA/sale-workflow/assets/17847432/5dbfddf3-9406-4ef3-9e2e-ef16968f1f68)

rafaelbn reviewed (approved):
/ocabot merge patch
